libs/libxx/libcxx.defs

```diff
@@ -71,6 +71,23 @@ libcxx/src/locale.cpp_CXXFLAGS += -Wno-shadow
 libcxx/src/filesystem/directory_iterator.cpp_CXXFLAGS += -Wno-shadow
 libcxx/src/filesystem/operations.cpp_CXXFLAGS += -Wno-shadow
 
+# The following warning was observed with icicle:knsh config.
+# Looking at the code in question, it seems harmless to ignore.
+#
+# Note: For some reasons, GCC -Wall enables a different set of warnings
+# for C and C++.
+#
+# References:
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=10604
+# https://github.com/gcc-mirror/gcc/blob/e54375d85d4aa5889869c2672158083b2106b623/gcc/c-family/c.opt#L1285-L1287
+#
+# CXX:  libcxx/src/condition_variable.cpp
+# libcxx/src/condition_variable.cpp: In member function 'void std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long int, std::__1::ratio<1, 1000000000> > >)':
+# libcxx/src/condition_variable.cpp:64:19: error: comparison of integer expressions of different signedness: 'std::__1::chrono::duration<long long int>::rep' {aka 'long long int'} and 'std::__1::__libcpp_numeric_limits<long unsigned int, true>::type' {aka 'long unsigned int'} [-Werror=sign-compare]
+#      if (s.count() < ts_sec_max)
+#          ~~~~~~~~~~^~~~~~~~~~~~
+libcxx/src/condition_variable.cpp_CXXFLAGS += -Wno-sign-compare
```

Comment:
libcxx is just one place which generate this warning. Since all major OS typedef time_t to 64bit signed integer, more and more code will generate the same warning, do you want to fix all of them?

Reply:
> libcxx is just one place which generate this warning. Since all major OS typedef time_t to 64bit signed integer, more and more code will generate the same warning, do you want to fix all of them?

of course i don't.
but we need to deal with these warnings for 32-bit time_t anyway.

> BTW, it's very ugly to add -Wno-xxx in a common Make.defs(not all compiler support -Wno-sign-compare), so please modify the code to remove the warning and submit the change to libcxx community.

i will do so sooner or later.

Reply:
> > BTW, it's very ugly to add -Wno-xxx in a common Make.defs(not all compiler support -Wno-sign-compare), so please modify the code to remove the warning and submit the change to libcxx community.
> 
> i will do so sooner or later.

https://reviews.llvm.org/D143140

Reply:
Thanks.

Reply:
look like the patch can't pass CI, @yamt ?

Reply:
@yamt could you update the patch as suggest in https://reviews.llvm.org/D143140?